CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKCommandBuffer.java

```diff
@@ -166,65 +167,43 @@ public void BeginRecordSecondary(int flags, long framebuffer, long renderPass, i
         inheritanceInfo.queryFlags(0);
         inheritanceInfo.pipelineStatistics(0);
 
-        BeginRecordSecondary(flags, inheritanceInfo);
-        
+        return BeginRecordSecondary(flags, inheritanceInfo);  
+    }
+	
+    public int FinishRecord() {
+        int ret;
+
+        ret = vkEndCommandBuffer(vkCommandBuffer);
+        if (VkFailed(ret)) { return ret; }
+
+        return ret;
     }
 	
-    public void FinishRecord(){
+    public void BeginRenderPass(long renderPass, long frameBuffer,
+                    int width, int height, int colorAttachmentCount, int depthAttachment,
+                    int contentsFlag) {
 
-        int err = vkEndCommandBuffer(vkCommandBuffer);
+        VkClearValue.Buffer clearValues = VkClearValue.calloc(colorAttachmentCount + depthAttachment);
 
-        if (err != VK_SUCCESS) {
-            throw new AssertionError("Failed to finish record command buffer: ");
+        for (int i=0; i<colorAttachmentCount; i++) {
+            clearValues.put(CVKUtils.getClearValueColor(new Vector3f(0,0,0)));
+        }
+        if (depthAttachment == 1) {
+            clearValues.put(CVKUtils.getClearValueDepth());
         }
+        clearValues.flip();
+
+        CVKCommandBuffer.this.BeginRenderPass(renderPass, frameBuffer, width, height,
```

Comment:
What the?  I think you can remove the CVKCommandBuffer.this.

Reply:
I dont know where that came from! Removed anyway